app/src/main/java/com/epfl/drawyourpath/pathDrawing/PathDrawingModel.kt

```diff
@@ -0,0 +1,122 @@
+package com.epfl.drawyourpath.pathDrawing
+
+import android.os.Handler
+import android.os.Looper
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
+import com.epfl.drawyourpath.path.Path
+import com.epfl.drawyourpath.path.Run
+import com.google.android.gms.maps.model.LatLng
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+
+class PathDrawingModel : ViewModel() {
+
+    private var resultingRun = Run(Path(), 0, 1)
+    private val _run: MutableLiveData<Run> = MutableLiveData(resultingRun)
+    private var startTime: Long? = null
+    private var pauseTime: Long? = null
+    private var pauseDuration: Duration = Duration.ZERO
+    private var lastRunUpdate: Long = 0

```

Comment:
Same remark for pause duration, like resultingRun

Reply:
for pause duration it is because at first the total duration of the pause is 0.